controllers/image_generator.go

```diff
@@ -225,200 +226,143 @@ func (r *ImageGenerator) createJobFromFile(jobFileName string) (*batchv1.Job, er
 				return nil, err
 			}
 			return latest, nil
-		} else {
-			return nil, err
 		}
+		return nil, err
 	}
 	igLogger.Info("Job file has been processed successfully and Job has been created")
 	return job, nil
 }
 
-func (r *ImageGenerator) deleteJobFromFile(jobFileName string, keepPods bool) error {
-	igLogger.Info("delete job from YAML file", "jobFileName", jobFileName)
-	yamlData, err := readJobYAML(jobFileName)
-	if err != nil {
-		return err
-	}
+// Method to delete job given job name and namespace
+func (r *ImageGenerator) deleteJob(jobName, namespace string) error {
+	igLogger.Info("Deleting Job", "jobName", jobName, "namespace", namespace)
 
-	job, err := parseJobYAML(yamlData)
+	job := &batchv1.Job{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      jobName,
+		Namespace: namespace,
+	}, job)
 	if err != nil {
 		return err
 	}
 
-	var deletePolicy metav1.DeletionPropagation
-	if keepPods {
-		deletePolicy = metav1.DeletePropagationOrphan // this leaves the job's pods around for tracking
-	} else {
-		deletePolicy = metav1.DeletePropagationBackground // this deletes the job's pods as well
-	}
+	// Set propagation policy to delete all the dependents
+	// This is required to delete the pods created by the job
+	// If the job has already been deleted, return nil
+	// This is to avoid the error "job.batch "jobName" not found"
+	// when the job has already been deleted
 
-	if err := r.client.Delete(context.TODO(), job, &client.DeleteOptions{PropagationPolicy: &deletePolicy}); err != nil {
+	deletePolicy := metav1.DeletePropagationBackground
+	if err := r.client.Delete(context.TODO(), job, &client.DeleteOptions{
+		PropagationPolicy: &deletePolicy,
```

Comment:
nit, you can skip setting the deletePolicy variable

Reply:
Do we want to keep the pod ?